src/aspire/reconstruction/mean.py

```diff
@@ -1,17 +1,262 @@
 import logging
+from functools import partial
 
 import numpy as np
+from scipy.linalg import norm
+from scipy.sparse.linalg import LinearOperator, cg
 
+from aspire import config
 from aspire.nufft import anufft
 from aspire.numeric import fft
 from aspire.operators import evaluate_src_filters_on_grid
-from aspire.reconstruction import Estimator, FourierKernel
-from aspire.volume import rotated_grids
+from aspire.reconstruction import Estimator, FourierKernel, FourierKernelMat
+from aspire.volume import Volume, rotated_grids
 
 logger = logging.getLogger(__name__)
 
 
-class MeanEstimator(Estimator):
+class WeightedVolumesEstimator(Estimator):
+    def __init__(self, weights, *args, **kwargs):
+        """
+        Weighted mean volume estimation.
+
+        This is best considered as an r x r matrix of volumes;
```

Comment:
I think this description needs to be revised. The r×r matrix of volumes here is the `FourierKernelMat`. The problem being solved here is the minimization given by eq. (14) in the paper, rewritten as the normal equations in eq. (20) and more compactly in eq. (23).

Reply:
Looks great. Thanks!